### PR TITLE
Remove feature manager

### PR DESCRIPTION
**Pull request by dlh01**
Having now spent time with a couple plugins created with `create-wordpress-plugin`, including the core plugin on a new production project, I'd like to propose that the current `Feature_Manager` class be removed.

I'm not necessarily opposed to the idea of having a bundled "manager"-type class. But I think the current implementation doesn't quite hit the mark for a few reasons. Among these:

1. Features are implementations of the interface provided by WP Type Extensions, and [a design goal of Type Extensions](https://github.com/alleyinteractive/wp-type-extensions/blob/a67070fc2d03422c5e24f83a211bffea7845ffa3/README.md#type-extensions) is to encourage "object composition, in particular the decorator pattern." But it isn't currently feasible to use decorators with feature classes in the plugin because features are given to the manager as unique class names, not objects. The current implementation is leaving possibilities provided by the design pattern on the table, I think.

2. The feature manager is having to do work that PHP could do itself: performing the type check on the passed interface and accepting named constructor parameters. IDEs aren't able to check parameter names and types, either.

3. It's impossible to assume that any of the features included in the plugin are going to run, since the `create_wordpress_plugin_features` filter implies that it's acceptable for extenders to remove any classes they want from the manifest. This takes an unexpected (IMO) degree of control out of the plugin developer's hands and makes plugin behavior less predictable.

With these points in mind, I suggest we take out the current feature manager, allow some experimentation to occur in individual plugins created with this template, and try with another approach later if there turns out to be demand.

### CODE REVIEW

File: plugin.php [hunk 1]
```diff
@@ -63,11 +65,8 @@ function () {
...
  */
 function main(): void {
-	// This should be an array with keys set to feature classnames and arguments.
-	$features = [
-		// Add initial features here.
-	];
-	$features = apply_filters( 'create_wordpress_plugin_features', $features );
-	Feature_Manager::add_features( $features );
+	// Add features here.
+	$plugin = new Group();
```
Comment (kevinfodness) [inline plugin.php:69]: Something that is lost by this approach is the ability to get access to the list of loaded features. Previously, you'd be able to do that using static methods on the feature manager class, but here, it's all wrapped up in the main function.

Some thoughts:

1. Is it important to be able to get access to a list of loaded features at all? Is it useful in tests or elsewhere in code?
2. This implementation also drops the filter - if we brought the filter back, would that be sufficient for "giving access" to the list of features, and implementers could store a reference to features at that point if they needed to?

An alternate proposal:

```suggestion
	$plugin = new Group(
		...apply_filters(
			'create_wordpress_plugin_features',
			[
				// Add initial features here by creating new instances of classes that implement the Feature interface.
			]
		)
	);
```

Comment (dlh01) [reply plugin.php:69]: > Something that is lost by this approach is the ability to get access to the list of loaded features. Previously, you'd be able to do that using static methods on the feature manager class, but here, it's all wrapped up in the main function.

That's true. But I'm somewhat skeptical of the value of that list.

> Some thoughts:
> 
> 1. Is it important to be able to get access to a list of loaded features at all? Is it useful in tests or elsewhere in code?

My question exactly 😄 I'm not sure that I see the use case yet.

> 2. This implementation also drops the filter - if we brought the filter back, would that be sufficient for "giving access" to the list of features, and implementers could store a reference to features at that point if they needed to?
> 
> An alternate proposal:

This version of the filter leaves me with the same questions that the current filter does. What's the value in allowing extenders to remove classes from the manifest of features? Doesn't that make it difficult for a plugin developer to predict the behavior of the plugin?

Similarly, what's the value in allowing extenders to add their own classes to the manifest? An extender could slip in any feature instance and have the plugin treat the feature like its own — with error handling, for example. It seems unusually permissive.

Comment (kevinfodness) [reply plugin.php:69]: [Alleyvate](https://github.com/alleyinteractive/wp-alleyvate) immediately comes to mind as a place where we _do_ want this list of features to be filtered. In that case, we're doing it by a different mechanism, but could convert it to be a simpler filter of all features vs. running filters on each individual feature being activated.

However, I'm not convinced of the need to make the filter part of what the plugin produces by default - implementers could always add in this functionality if they need it.

**Review by kevinfodness — APPROVED** [review]
🍣

File: plugin.php (hunk 1 above)
Comment (renatonascalves) [inline plugin.php:69]: Can we add a comment line to [this documentation](https://github.com/alleyinteractive/create-wordpress-plugin/tree/90fd76a78a74426712c2ce7ba601ca30344e82ba/src/features#adding-a-feature)? As to educate people on how to add new features.

Comment (renatonascalves) [reply plugin.php:69]: > However, I'm not convinced of the need to make the filter part of what the plugin produces by default - implementers could always add in this functionality if they need it.

I'd agree.

**Review by renatonascalves — APPROVED** [review]
🗡️

File: plugin.php (hunk 1 above)
Comment (kevinfodness) [reply plugin.php:69]: There's an example in the README of how to do this, which fills the documentation need, imo.

**Pull request merged**